packages/toolpad-app/src/components/AppEditor/AppEditorShell.tsx

```diff
@@ -109,6 +111,28 @@ export interface ToolpadAppShellProps {
 
 export default function AppEditorShell({ appId, ...props }: ToolpadAppShellProps) {
   const domLoader = useDomLoader();
+  const dom = useDom();
+
+  const navigate = useNavigate();
+  const { state: locationState } = useLocation();
+
+  const redirectRef = React.useRef<boolean>(false);
+
+  /* Redirect to first page (if it exists) if (any of the following):
+   * 1. first visit to the Editor
+   * 2. if being redirected from a delete operation
+   */
+  React.useEffect(() => {
+    const app = appDom.getApp(dom);
+    const { pages = [] } = appDom.getChildNodes(dom, app);
+    if (
+      pages?.length &&
```

Comment:
Two Side note: 
- no need to use optional chaining `?` if it can't be `null`/`undefined`
- Always use explicit boolean check for `[].length`: `pages.length > 0`